tools/run-integration-tests.py

```diff
@@ -0,0 +1,182 @@
+                if proposed:
+                    env["UACLIENT_BEHAVE_ENABLE_PROPOSED"] = "1"
+                else:
+                    env["UACLIENT_BEHAVE_BUILD_PR"] = "1"
+
+                # Inject tokens from credentials
+                for t in token:
+                    envvar = TOKEN_TO_ENVVAR[t]
+                    env[envvar] = credentials["token"].get(envvar)
+
+                # Inject cloud-specific variables from credentials
+                for cloud in ("azure", "aws", "gcp"):
+                    if cloud in p:
+                        for envvar in credentials[cloud]:
+                            env[envvar] = credentials[cloud].get(envvar)
+                # Tox command itself
+                command = "tox -e behave-{}-{}".format(
+                    p, series_version
+                ).split()
+
+                # Wip
+                if wip:
+                    command.extend(["--tags='wip'", "--stop"])
+
+                commands.append((command, env))
+
+    return commands
+
+
+@click.command()
+@click.option(
+    "-p",
+    "--platform",
+    type=click.Choice(PLATFORMS),
+    default=PLATFORMS,
+    multiple=True,
+    help="Platform to run the tests on (requires credentials file for the clouds)",
+)
+@click.option(
+    "-s",
+    "--series",
+    type=click.Choice(SERIES_TO_VERSION.keys()),
+    default=SERIES_TO_VERSION.keys(),
+    multiple=True,
+    help="Series to run the tests on",
+)
+@click.option(
+    "--token",
+    "-t",
+    type=click.Choice(TOKEN_TO_ENVVAR.keys()),
+    default=TOKEN_TO_ENVVAR.keys(),
+    multiple=True,
+    help="Tokens to use for the tests (require tokens in the credentials file)",
+)
+@click.option(
+    "--proposed",
+    type=bool,
+    default=False,
+    is_flag=True,
+    help="Run tests based on existing packages in the Proposed pocket",
+)
+@click.option(
+    "--check-version",
+    type=str,
+    help="Check for a specific version in the tests",
+)
+@click.option(
+    "--credentials-path",
+    "-c",
+    type=click.Path(dir_okay=False),
+    default="tools/ua-test-credentials.yaml",
+    help="Path to the file containing the credentials to run the test",
+)
+@click.option(
+    "-w",
+    "--wip",
+    type=bool,
+    default=False,
+    is_flag=True,
+    help="Run only tests with the @wip decorator",
+)
+def run_tests(
+    platform, series, proposed, check_version, token, credentials_path, wip
+):
+    commands = build_commands(
+        platform, series, proposed, check_version, token, credentials_path, wip
+    )
+    current_datetime = datetime.now().strftime(format="%b-%d-%H%M")
+    output_dir = "test-results-{}".format(current_datetime)
+    os.makedirs(output_dir, exist_ok=True)
+    processes = []
+    for command, env in commands:
+        print("Running {}".format(command))
+        with open(
+            "test-results/{}.txt".format(command[2]), "wb"
+        ) as result_file:
```

Comment:
Thanks for adding this! I think it might be nice to have the datetime as a subdir under test-results so as not to pollute the root level of the project. Also we need to make sure we use the right dir later in `open`.
```suggestion
    current_datetime = datetime.now().strftime(format="%b-%d-%H%M")
    output_dir = "test-results/{}".format(current_datetime)
    os.makedirs(output_dir, exist_ok=True)
    processes = []
    for command, env in commands:
        print("Running {}".format(command))
        with open(
            "{}/{}.txt".format(output_dir, command[2]), "wb"
        ) as result_file:
```

And then later on, lets use the datetime again in the tarball name and for the tarball contents
```
        filename = "test-results-{}.tar.gz".format(current_datetime)
        with tarfile.open(filename, "w:gz") as results:
            results.add(output_dir, arcname="test-results/")
```

Reply:
Indeed